addons/point_of_sale/static/src/js/screens.js

```diff
@@ -1866,6 +1870,14 @@ var PaymentScreenWidget = ScreenWidget.extend({
     // handle both keyboard and numpad input. Accepts
     // a string that represents the key pressed.
     payment_input: function(input) {
+        var paymentline = this.pos.get_order().selected_paymentline;
+
+        // disable changing amount on paymentlines with running or done payments on a payment terminal
+        if (this.payment_interface &&
+            ['done', 'force_done', 'waitingCard', 'waiting', 'reversing', 'reversed'].includes(paymentline.get_payment_status())) {
```

Comment:
Could be easier to check if it's not `pending` or `retry` if we ever want to add new steps during the payment

Reply:
Yeah, it's shorter as well. I changed it.